pkg/app/api/grpcapi/piped_api.go

```diff
@@ -968,12 +968,51 @@ func (a *PipedAPI) UpdateApplicationConfigurations(ctx context.Context, req *pip
 			return nil, err
 		}
 	}
+
+	opts := datastore.ListOptions{
+		Limit: 1,
+		Filters: []datastore.ListFilter{
+			{
+				Field:    "ProjectId",
+				Operator: datastore.OperatorEqual,
+				Value:    projectID,
+			},
+			{
+				Field:    "Deleted",
+				Operator: datastore.OperatorEqual,
+				Value:    false,
+			},
+		},
+	}
+	envs, err := a.environmentStore.ListEnvironments(ctx, opts)
+	if err != nil {
+		a.logger.Error("failed to get environments", zap.Error(err))
+		return nil, status.Error(codes.Internal, "Failed to get environments")
+	}
+	envsMap := make(map[string]string, len(envs))
+	for _, env := range envs {
+		// It is assumed that the env name is kept unique by the user.
+		envsMap[env.Name] = env.Id
+	}
+
 	for _, appInfo := range req.Applications {
+		envID := ""
+		if appInfo.EnvName != "" {
+			var ok bool
+			envID, ok = envsMap[appInfo.EnvName]
+			if !ok {
+				a.logger.Error("unknown environment name given",
+					zap.String("env", appInfo.EnvName),
+					zap.Error(err),
+				)
+				return nil, status.Errorf(codes.InvalidArgument, "Unknown environment name %q given", appInfo.EnvName)
+			}
+		}
 		updater := func(app *model.Application) error {
 			app.Name = appInfo.Name
 			app.Labels = appInfo.Labels
 			app.Description = appInfo.Description
-			// TODO: Enable to update env via PipedAPI
+			app.EnvId = envID
```

Comment:
This will override the existing configured EnvID of the app that has not been migrated to use EnvName in the app config file?

Reply:
Yes. That is what we expect